app/helpers/bookings_helper.rb

```diff
@@ -1,13 +1,37 @@
 module BookingsHelper
   def get_cost(booking)
-    passengers = booking["passengers_attributes"]
-    flight = Flight.find_by_id(booking["flight_id"])
+    passengers = booking.passengers
+    flight = booking.flight
     cost = 0
-    passengers.each do |_key, value|
-      cost += flight.economy if value["cabin"] == "Economy"
-      cost += flight.first if value["cabin"] == "First"
-      cost += flight.business if value["cabin"] == "Business"
+    passengers.each do |passenger|
+      cost += flight.economy if passenger.cabin == "Economy"
+      cost += flight.first if passenger.cabin == "First"
+      cost += flight.business if passenger.cabin == "Business"
+    end
+
+    passengers.each_with_index do |passenger|
+      passenger.cost = flight.economy if passenger.cabin == "Economy"
+      passenger.cost = flight.first if passenger.cabin == "First"
+      passenger.cost = flight.business if passenger.cabin == "Business"
     end
     cost
   end
+
+  def payment_url(booking)
+    values = {
+      business: "oluosiname-facilitator@yahoo.com",
+      cmd: "_cart",
+      upload: 1,
+      return:  success_bookings_url(:id => booking.id),
+      invoice: booking.ref
+    }
+
+    booking.passengers.each_with_index do |passenger, index|
+      values.merge!(
```

Comment:
Use `values["amount_#{index + 1}"] = passenger.cost; values["item_name_#{index + 1}"] = passenger.first_name` instead of `values.merge!(
        "amount_#{index + 1}" => passenger.cost,
        "item_name_#{index + 1}" => passenger.first_name
      )`.